PlayFab/MpsAgent
# Added logging for game not found bug

The bug ADO Task in question:

- [[VmAgent] Process-based servers don't capture logs on bad startup command](https://microsoft.visualstudio.com/Xbox/_sprints/taskboard/AGS%20MP%20Hosting/Xbox/22-4W24%20(Ends%206-10)/2W24%20(Ends%206-10)?workitem=33466233)

In summary, if a user is testing their game.exe but the file is not found, there is no logging done currently to make the user aware of the issue. This PR focuses on Windows-based processes.

I've added functionality to a separate CollectLogs() method within ProcessRunner.cs that is only called if the process failed to start. If the game starts successfully, no log file is created in the destination directory. If the game fails, a file with the exception message is created in the destination directory.

To test:
1. You can use the MultiplayerSettings.json used in MpsSamples. Edit "OutputFolder" path with a local folder path of your choosing. Edit "StartGameCommand" to have a game.exe that does not exist (i.e. nonExistent.exe).
2. Delete LocalMultiplayerAgentPublishFolder.
3. Choose LocalMultiplayerAgent as your Startup Project. 
4. Create breakpoint at try{} catch{} blocks for _processWrapper.Start(...) on line 65 and 70, respectively, within ProcessRunner.cs.
5. Select LocalMultiplayerAgent as Build, and begin Debug.
6. Check file called PF_ConsoleLogs.txt was created in destination directory with helpful log that tells you "The system cannot find the file specified.".

Please let me know if you have any recommendations or better courses of actions. Just trying to get my first PR in and understanding this new environment :)
